### PR TITLE
machines: Auto-detect spice capability when creating a VM

### PR DESCRIPTION
**Pull request by martinpitt**
QEMU does not have SPICE enabled on all architectures. Use QMP's
`query-spice` command to detect the capability, and don't create a VM
with spice if it is not supported, as that will fail.

If supported, `query-spice` will answer with something like

    {"return": {"migrated": false, "enabled": false, "mouse-mode": "client"}}

If not supported, it will look like

    {"error": {"class": "CommandNotFound", "desc": "The command query-spice has not been found"}}

Also make the create_machine.sh script `set -e` and fix a whitespace error.

https://bugzilla.redhat.com/show_bug.cgi?id=1563418

### CODE REVIEW

**Comment by martinpitt** [comment]
Note: We don't currently have integration tests that start the VM, due to issue #8385. So this is tricky to validate automatically right now. I've tested it manually on x86_64 and i386 and spice still works fine. I'll ask Gustavo on https://bugzilla.redhat.com/show_bug.cgi?id=1563418 to validate this on ppc64el where spice does not work.

File: pkg/machines/scripts/create_machine.sh [hunk 1]
```diff
@@ -13,12 +13,25 @@ vmExists(){
...
         exit $1
     fi
 }
 
+spiceSupported(){
+    # map system architecture to qemu-system-* command
+    ARCH=$(uname -m)
+    case "$ARCH" in
+        i?86) QEMU=qemu-system-i386 ; break ;;
+        ppc64*) QEMU=qemu-system-ppc64 ; break ;;
```
Comment (suomiy) [inline pkg/machines/scripts/create_machine.sh]: I think break is not needed or valid here

File: pkg/machines/scripts/create_machine.sh [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/bin/sh
+set -eu -o noglob
```
[the remark this answers is not in the document]
Comment (martinpitt) [reply pkg/machines/scripts/create_machine.sh]: It works quite well on the "everything succeeds" case, but indeed that would miss the cleanup on failures. Thanks for pointing out! Fixed. I split that out into a separate commit now.

File: pkg/machines/scripts/create_machine.sh (hunk 1 above)
Comment (martinpitt) [reply pkg/machines/scripts/create_machine.sh]: Eww, indeed; it's a no-op (so I didn't notice it when testing on 32 bit), but it leaves an ugly `break: only meaningful in a `for', `while', or `until' loop` message in the logs. Fixed.

File: pkg/machines/scripts/create_machine.sh [hunk 3]
```diff
@@ -13,12 +13,25 @@ vmExists(){
 }
 
 endIfFailed(){
-	if [ $1 -ne 0 ]; then
-	    rm -f "$XMLS_FILE"
+    if [ $1 -ne 0 ]; then
```
Comment (suomiy) [inline pkg/machines/scripts/create_machine.sh]: this if is not necessary here anymore. Also the name of the function could be something like `finish()` now

Comment (martinpitt) [reply pkg/machines/scripts/create_machine.sh]: Good point. Fixed.

**Comment by suomiy** [comment]
looks fine, thanks

**Comment by martinpitt** [comment]
@larskarlitski this was reviewed already, can you please land?

**Comment by larskarlitski** [comment]
> @larskarlitski this was reviewed already, can you please land?

Sure :)

**Pull request merged**